Tested the PR with the following steps:
#### On master:
```bash
python3 feature_assumeutxo.py
...
TestFramework (INFO): Tests successful
```
Modified validation.cpp (line 6237, function MaybeCompleteSnapshotValidation):
```diff
- return SnapshotCompletionResult::SUCCESS;
+ return SnapshotCompletionResult::SKIPPED;
```
This should avoid the clean-up of the snapshot at startup. But the test is not affected:
```bash
python3 feature_assumeutxo.py
...
TestFramework (INFO): Tests successful
```
#### On PR branch:
```bash
python3 feature_assumeutxo.py
...
TestFramework (INFO): Tests successful
```
Modified validation.cpp as before. This time the test fails as it should:
```bash
python3 feature_assumeutxo.py
...
TestFramework (ERROR): Assertion failed
...
AssertionError: [node 1] Expected messages "['cleaning up unneeded background chainstate']" does not partially match log:
...
```

#### NOTE
The change that I did to validation.cpp causes the "validation_chainstatemanager_tests" unit test to fail, so that line is covered. I don't know if there is a simple way to change the code that would result in success in all tests in master. Having the cleaning code covered in a functional test seems like a good idea anyway.

tACK dfd0a76fc1b46ddd1f4dab388ec2ca7fc3f76a2b

--- a/test/functional/feature_assumeutxo.py
+++ b/test/functional/feature_assumeutxo.py
@@ -9,6 +9,7 @@ to a hash that has been compiled into bitcoind.
 The assumeutxo value generated and used here is committed to in
 `CRegTestParams::m_assumeutxo_data` in `src/kernel/chainparams.cpp`.
 """
+import contextlib
 from shutil import rmtree
 
 from dataclasses import dataclass
@@ -337,6 +338,22 @@ class AssumeutxoTest(BitcoinTestFramework):
         assert 'NETWORK' not in node_services
         assert 'NETWORK_LIMITED' in node_services
 
+    @contextlib.contextmanager
+    def assert_disk_cleanup(self, node, assumeutxo_used):
+        """
+        Ensure an assumeutxo node is cleaning up the background chainstate
+        """
+        msg = []
+        if assumeutxo_used:
+            # Check that the snapshot actually existed before restart
+            assert (node.datadir_path / node.chain / "chainstate_snapshot").exists()
+            msg = ["cleaning up unneeded background chainstate"]
+
+        with node.assert_debug_log(msg):
+            yield
+
+        assert not (node.datadir_path / node.chain / "chainstate_snapshot").exists()
+
     def run_test(self):
         """
         Bring up two (disconnected) nodes, mine some new blocks on the first,
@@ -644,7 +661,8 @@ class AssumeutxoTest(BitcoinTestFramework):
         for i in (0, 1):
             n = self.nodes[i]
             self.log.info(f"Restarting node {i} to ensure (Check|Load)BlockIndex passes")
-            self.restart_node(i, extra_args=self.extra_args[i])
+            with self.assert_disk_cleanup(n, i == 1):
+                self.restart_node(i, extra_args=self.extra_args[i])
 
             assert_equal(n.getblockchaininfo()["blocks"], FINAL_HEIGHT)
 
@@ -721,7 +739,8 @@ class AssumeutxoTest(BitcoinTestFramework):
         for i in (0, 2):
             n = self.nodes[i]
             self.log.info(f"Restarting node {i} to ensure (Check|Load)BlockIndex passes")
-            self.restart_node(i, extra_args=self.extra_args[i])
+            with self.assert_disk_cleanup(n, i == 2):
+                self.restart_node(i, extra_args=self.extra_args[i])
 
             assert_equal(n.getblockchaininfo()["blocks"], FINAL_HEIGHT)
 

